x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -28,22 +28,27 @@ import { getChromeLogLocation } from '../paths';
 import { puppeteerLaunch } from '../puppeteer';
 import { args } from './args';
 
-type binaryPath = string;
+type BinaryPath$ = Rx.Observable<string>;
 type BrowserConfig = CaptureConfig['browser']['chromium'];
 type ViewportConfig = CaptureConfig['viewport'];
 
 export class HeadlessChromiumDriverFactory {
-  private binaryPath: binaryPath;
+  private binaryPath?: string;
+  private binaryPath$: BinaryPath$;
   private captureConfig: CaptureConfig;
   private browserConfig: BrowserConfig;
   private userDataDir: string;
   private getChromiumArgs: (viewport: ViewportConfig) => string[];
 
-  constructor(binaryPath: binaryPath, logger: LevelLogger, captureConfig: CaptureConfig) {
-    this.binaryPath = binaryPath;
+  constructor(binaryPath$: BinaryPath$, captureConfig: CaptureConfig, logger: LevelLogger) {
+    this.binaryPath$ = binaryPath$;
```

Comment:
This constructor is called in setup -- getting access to the binary path string synchronously would slow down setup.